src/main/java/com/example/UserModule/service/serviceImpl/UserServiceImplementation.java

```diff
@@ -175,7 +175,7 @@ public SignInResponseDto signIn(SignInDto signInDto, String ip, HttpServletReque
     } catch (GeoIp2Exception e) {
       throw new RuntimeException(e);
     }
```

Comment:
**Suggestion:** Consider handling the RuntimeException in a more graceful way. Throwing a RuntimeException directly is not a good practice. You should create a custom exception and throw that instead.
```suggestion
    } catch (GeoIp2Exception e) {
      throw new CustomException("GeoIP2 exception occurred", e);
    }
```